tests/matrix_free/matrix_vector_hessians_cells.cc

```diff
@@ -218,13 +218,13 @@ test_hessians(const dealii::FE_Poly<dim>                    &fe,
 
   // compare solutions of matrix vector product
   {
-    dst2 -= dst;
+    dst -= dst2;
 
     double error = 0.;
-    if (dst.l2_norm() > 0)
-      error = dst2.l2_norm() / dst.l2_norm();
+    if (dst2.l2_norm() > 0)
+      error = dst.l2_norm() / dst2.l2_norm();
     else
-      error = dst2.l2_norm();
+      error = dst.l2_norm();
```

Comment:
I changed this test slightly: We might indeed have roundoff errors in some cases with this change (as Hessians are done in the collocation space). In this test, I observed that `dst2` (reference result) was exactly zero, whereas `dst` contained roundoff errors around `1e-16`. The previous normalization would make the result `1`, whereas it in fact should be small. I admit this is not the final solution as we should probably not even try to normalize in the first place, but I think it is more robust than before and I did not have the time to think more about the test.

Reply:
I don't remember the precise reason for the normalization back when I wrote those `matrix_vector_hessians_*` tests but I think it was to make them more sensitive to make them trigger for wrong results for every polynomial degree. We should reevaluate this for all of these tests.